modules/vmfcore/java/Variant.java

```diff
@@ -0,0 +1,236 @@
+		//nativeObj = n_Variant (str);
+	}
+	
+	public void copy (Variant other)
+	{
+		n_copy (nativeObj, other.nativeObj);
+	}
+	
+	public Variant setTo (int value)
+	{
+		Variant retVal = new Variant (n_setTo (nativeObj, value));
+		return retVal;
+	}
+	
+	public Variant setTo (float value)
+	{
+		Variant retVal = new Variant (n_setTo (nativeObj, value));
+		return retVal;
+	}
+	
+	public Variant setTo (int[] array)
+	{
+		Variant retVal = new Variant (n_setTo (nativeObj, array));
+		return retVal;
+	}
+	
+	public Variant setTo (float[] array)
+	{
+		Variant retVal = new Variant (n_setTo (nativeObj, array));
+		return retVal;
+	}
+	
+	public Variant setTo (String str)
+	{
+		Variant retVal = new Variant (n_setTo (nativeObj, str));
+		return retVal;
+	}
+	
+	public Variant setTo (Variant other)
+	{
+		Variant retVal = new Variant (n_setTo (nativeObj, other.nativeObj));
+		return retVal;
+	}
+	
+	public boolean equals (Variant other)
+	{
+		return n_equals (nativeObj, other.nativeObj);
+	}
+	
+	public boolean unequals (Variant other)
+	{
+		return n_unequals (nativeObj, other.nativeObj);
+	}
+	
+	public String toString ()
+	{
+		return n_toString (nativeObj);
+	}
+	
+	public void fromString (int type, String str)
+	{
+		n_fromString (nativeObj, type, str);
+	}
+	
+	public int getType ()
+	{
+		return n_getType (nativeObj);
+	}
+	
+	public boolean isEmpty ()
+	{
+		return n_isEmpty (nativeObj);
+	}
+	
+	public String getTypeName ()
+	{
+		return n_getTypeName(nativeObj);
+	}
+	
+	public void clear ()
+	{
+		if (nativeObj != 0)
+		    n_delete (nativeObj);
+		
+		nativeObj = n_Variant ();
+	}
+	
+	public void convertTo (int type)
+	{
+		n_convertTo (nativeObj, type);
+	}
+	
+	public static String typeToString (int type)
+	{
+		return n_typeToString (type);
+	}
+
+	public static int typeFromString (String fieldType)
+	{
+		return n_typeFromString (fieldType);
+	}
+	
+	public static boolean isConvertible (int srcType, int tarType)
+	{
+		return n_isConvertible (srcType, tarType);
+	}
+	
+	public static String base64Encode (vmfRawBuffer buf)
+	{
+		return n_base64Encode (buf.byteBuf, buf.size);
+	}
+	
+	static vmfRawBuffer base64Decode (String base64Str)
+	{
+		vmfRawBuffer buf = new vmfRawBuffer ();
+		int size[] = new int[1];
+		buf.byteBuf = n_base64Decode (base64Str, size);
+		buf.size = size[0]; 
+		return buf;
+	}
+	
+    @Override
+    protected void finalize () throws Throwable 
+    {
+    	if (nativeObj != 0)
+    	{
+            n_delete (nativeObj);
+            nativeObj = 0;
+    	}
+        super.finalize();
+    }
+	
+	private native long n_Variant ();
+	/*private native long n_Variant (final int value);
+	private native long n_Variant (final float value);
+	private native long n_Variant (int[] array);
+	private native long n_Variant (float[] array);
+	private native long n_Variant (String str);*/
+	private native static void n_copy (long nativeObj, long other);
+	private native static long n_setTo (long nativeObj, int value);
```

Comment:
`setTo` should assign a value to the current object instead of creating a new one!
So it should be `void`